QuEST/src/QuEST_validation.c

```diff
@@ -679,6 +681,29 @@ void validateDiagonalOp(Qureg qureg, DiagonalOp op, const char* caller) {
     QuESTAssert(qureg.numQubitsRepresented == op.numQubits, E_MISMATCHING_QUREG_DIAGONAL_OP_SIZE, caller);
 }
 
+void validateIsIsingSpinHamiltonian(PauliHamil hamil, const char* caller){
+
+	int numQubits = hamil.numQubits;
+
+	for(int term_i = 0; term_i < hamil.numSumTerms; ++term_i){
+
+		qreal coeff = hamil.termCoeffs[term_i];
+		if(coeff == 0)
+			continue;
+
+		int num_zs = 0;
+		for(int qbit_i = 0; qbit_i < numQubits; ++qbit_i){
+
+			if(hamil.pauliCodes[term_i*numQubits+qbit_i] == PAULI_I)
+				continue;
+
+			QuESTAssert(hamil.pauliCodes[term_i*numQubits+qbit_i] == PAULI_Z, E_NOT_A_DIAGONAL_OPERATOR, caller);
+			QuESTAssert(++num_zs<=2, E_NOT_A_DIAGONAL_OPERATOR, caller);
```

Comment:
Why restrict to two qubit Z strings? Any number of qubits yields a diagonal operator; this function can hence be more general